dispatchers.go

```diff
@@ -196,10 +207,15 @@ func dispatchCopy(b *Builder, args []string, attributes map[string]bool, flagArg
 		case strings.HasPrefix(arg, "--from="):
 			from = strings.TrimPrefix(arg, "--from=")
 		default:
-			return fmt.Errorf("COPY only supports the --chmod=<permissions> --chown=<uid:gid> and the --from=<image|stage> flags")
+			return fmt.Errorf("COPY only supports the --chmod=<permissions>, --chown=<uid:gid>, --from=<image|stage>, and the --keep-ownership flags")
 		}
 	}
-	b.PendingCopies = append(b.PendingCopies, Copy{From: from, Src: args[0:last], Dest: dest, Download: false, Chown: chown, Chmod: chmod})
+
+	if chown != "" && preserveOwnership {
```

Comment:
Wy doesn't this return an error?
Shoulding --chown=foobar --keep-ownership conflict?

Reply:
I did change it, now an error is returned.